src/main/scala/viper/gobra/frontend/info/implementation/typing/ghost/GhostMiscTyping.scala

```diff
@@ -258,12 +258,39 @@ trait GhostMiscTyping extends BaseTyping { this: TypeInfoImpl =>
   }
 
   implicit lazy val wellDefSpec: WellDefinedness[PSpecification] = createWellDef {
-    case PFunctionSpec(pres, preserves, posts, _) =>
+    case n@ PFunctionSpec(pres, preserves, posts, terminationMeasures, _) =>
       pres.flatMap(assignableToSpec) ++ preserves.flatMap(assignableToSpec) ++ posts.flatMap(assignableToSpec) ++
-      preserves.flatMap(e => allChildren(e).flatMap(illegalPreconditionNode)) ++ 
-      pres.flatMap(e => allChildren(e).flatMap(illegalPreconditionNode))
+      preserves.flatMap(e => allChildren(e).flatMap(illegalPreconditionNode)) ++
+      pres.flatMap(e => allChildren(e).flatMap(illegalPreconditionNode)) ++
+      terminationMeasures.flatMap(wellDefTerminationMeasure) ++
+      // if has conditional clause, all clauses must be conditional
+      // can only have one non-conditional clause
+      error(n, "Specificaions can either contain one non-conditional termination measure or multiple conditional-termination measures.", terminationMeasures.length > 1 && !terminationMeasures.forall(isConditional)) ++
```

Comment:
```suggestion
      error(n, "Specifications can either contain one non-conditional termination measure or multiple conditional-termination measures.", terminationMeasures.length > 1 && !terminationMeasures.forall(isConditional)) ++
```

Reply:
why aren't these two errors applicable to loop spec?

Reply:
Because the loopspec has an `Option[PTerminationMeasure]` instead of a `Vector[PTerminationMeasure]`, which is the case for function spec. However, I forgot to check that the termination measure of the loop is not conditional.